### DISCUSSION TITLE
How to Override or Modify HTTP Status Code checks

### DISCUSSION THREAD

**Discussion by aplatypus** [testingisdocumenting/webtau#1152] [Q&A] [answered]
* I have a legacy end-point that returns 200 (OK) for a POST call.
* I want to permit/specify valid values of at least: 200 or 201, on this operation.
* More generally, what is the most effective method to enhance or override automatic checks like the Status Code on REST calls?

* I've been looking over the on-line pages, haven't found it yet.  I do not want to remove all validation, just add an alternative value to an existing check.

**Comment by MykolaGolubyev** [comment, marked as answer]
Hello, 
looks like docs for it a bit hidden, will think where to put them so users discover them easier
https://testingisdocumenting.org/webtau/HTTP/CRUD#implicit-statuscode-check

```
http.post("/customers/${id}", ....) {
    statusCode.should == 200
}
```

```
http.post("/customers/" + id, ((header, body) -> {
    header.statusCode.should(equal(200));
}));
```

**Reply by aplatypus** [reply]
* Thanks for that, these things are now working to match the existing version.
* One point though for Java I needed to do this:
```
        WebTauDsl.http.post(  callUrl,  restCallHeader, restCallBody, ((header, body) -> 
        {
            header.statusCode().should( equal(200) );
                  :
	}
``` 
   * The `statusCode()` is a method call. Quite neat nevertheless.

**Reply by MykolaGolubyev** [reply]
Ah right, upcoming version removes () and marks it deprecated but will still work. And updated docs around implicit status code check. 
thank you for the feedback!
